### PR TITLE
8329471: Remove GTK2

### PR DESCRIPTION
**Pull request by azvegint**
GTK2 support for Swing/AWT was deprecated for removal in JDK 21. 

It's being removed because no platform that supports the JDK needs it.

Manual and automated testing is green.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8337379](https://bugs.openjdk.org/browse/JDK-8337379) to be approved

### Issues
 * [JDK-8329471](https://bugs.openjdk.org/browse/JDK-8329471): Remove GTK2 (**Bug** - P3)
 * [JDK-8337379](https://bugs.openjdk.org/browse/JDK-8337379): Remove GTK2 (**CSR**)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20378/head:pull/20378` \
`$ git checkout pull/20378`

Update a local copy of the PR: \
`$ git checkout pull/20378` \
`$ git pull https://git.openjdk.org/jdk.git pull/20378/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20378`

View PR using the GUI difftool: \
`$ git pr show -t 20378`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20378.diff">https://git.openjdk.org/jdk/pull/20378.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20378#issuecomment-2256182507)

### CODE REVIEW

File: src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java
Comment (kumarabhi006) [inline src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java:1]: Should we remove this comment too ?

`static final int DEFAULT_TOGGLE_MENU_ITEM_SIZE = 12; // For pre-gtk2.4`

File: src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java [hunk 1]
```diff
@@ -500,12 +491,6 @@ public static GtkVersions getEnabledGtkVersion() {
                 new GetPropertyAction("jdk.gtk.version"));
         if (version == null) {
```
Comment (kumarabhi006) [inline src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java:492]: `null` check for version can be removed as if version is not 3.... `GtkVersions.ANY` is returned.

File: src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java [hunk 2]
```diff
@@ -500,12 +491,6 @@ public static GtkVersions getEnabledGtkVersion() {
...
             return GtkVersions.ANY;
-        } else if (version.startsWith("2")) {
-            if (!gtk2WarningIssued) {
-                System.err.println(GTK2_DEPRECATION_MESSAGE);
-                gtk2WarningIssued = true;
-            }
-            return GtkVersions.GTK2;
         } else if("3".equals(version) ){
```
Comment (kumarabhi006) [inline src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java:494]: spacing
```suggestion
        } else if ("3".equals(version)) {
```

File: src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java [hunk 3]
```diff
@@ -59,21 +59,21 @@ class GTKIconFactory {
...
 
     private static DelegatingIcon getIcon(String methodName) {
         DelegatingIcon result = iconsPool.get(methodName);
         if (result == null) {
-            if (methodName == TREE_COLLAPSED_ICON ||
-                methodName == TREE_EXPANDED_ICON)
+            if (TREE_COLLAPSED_ICON.equals(methodName) ||
+                    TREE_EXPANDED_ICON.equals(methodName))
             {
                 result = new SynthExpanderIcon(methodName);
```
Comment (kumarabhi006) [inline src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java:72]: blank lines can be removed.

File: src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java [hunk 4]
```diff
@@ -59,21 +59,21 @@ class GTKIconFactory {
...
+    private final static Map<String, DelegatingIcon> iconsPool =
             Collections.synchronizedMap(new HashMap<String, DelegatingIcon>());
 
     private static DelegatingIcon getIcon(String methodName) {
         DelegatingIcon result = iconsPool.get(methodName);
         if (result == null) {
-            if (methodName == TREE_COLLAPSED_ICON ||
-                methodName == TREE_EXPANDED_ICON)
+            if (TREE_COLLAPSED_ICON.equals(methodName) ||
+                    TREE_EXPANDED_ICON.equals(methodName))
```
Comment (kumarabhi006) [inline src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java:69]: can wrap the line before || operator

```suggestion
            if (TREE_COLLAPSED_ICON.equals(methodName)
                 || TREE_EXPANDED_ICON.equals(methodName))
```

File: test/jdk/java/awt/Gtk/GtkVersionTest/GtkVersionTest.java
Comment (kumarabhi006) [inline test/jdk/java/awt/Gtk/GtkVersionTest/GtkVersionTest.java:1]: Should we modify summary as well ?

copyright year and bug id needs to update.

File: test/jdk/java/awt/Robot/HiDPIScreenCapture/ScreenCaptureGtkTest.java
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/HiDPIScreenCapture/ScreenCaptureGtkTest.java:1]: copyright year and bug id needs to update.

File: test/jdk/javax/swing/LookAndFeel/8145547/DemandGTK.java
Comment (kumarabhi006) [inline test/jdk/javax/swing/LookAndFeel/8145547/DemandGTK.java:1]: copyright year and bug id needs to update.

**Comment by kumarabhi006** [comment]
Other references related to gtk2 here

1. [DesktopGtkLoadTest](https://github.com/openjdk/jdk/blob/574358df3b4891e953f84814c64ac651ea37755a/test/jdk/java/awt/Desktop/DesktopGtkLoadTest/DesktopGtkLoadTest.java#L56)
2. [DemandGTK3.sh](https://github.com/openjdk/jdk/blob/574358df3b4891e953f84814c64ac651ea37755a/test/jdk/javax/swing/LookAndFeel/8145547/DemandGTK3.sh#L28)

BugId and copyright year also can be updated.

File: src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java
Comment (azvegint) [reply src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKIconFactory.java:1]: Updated.

File: src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java (hunk 2 above)
Comment (azvegint) [reply src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java]: Updated

File: src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java (hunk 1 above)
Comment (azvegint) [reply src/java.desktop/unix/classes/sun/awt/UNIXToolkit.java]: Updated.

File: test/jdk/java/awt/Gtk/GtkVersionTest/GtkVersionTest.java
Comment (azvegint) [reply test/jdk/java/awt/Gtk/GtkVersionTest/GtkVersionTest.java:1]: added clarifying comment instead

**Comment by kumarabhi006** [comment]
> Other references related to gtk2 here
> 
> 1. [DesktopGtkLoadTest](https://github.com/openjdk/jdk/blob/574358df3b4891e953f84814c64ac651ea37755a/test/jdk/java/awt/Desktop/DesktopGtkLoadTest/DesktopGtkLoadTest.java#L56)


Did you miss to update this file?

**Comment by azvegint** [comment]
> > Other references related to gtk2 here
> > 
> > 1. [DesktopGtkLoadTest](https://github.com/openjdk/jdk/blob/574358df3b4891e953f84814c64ac651ea37755a/test/jdk/java/awt/Desktop/DesktopGtkLoadTest/DesktopGtkLoadTest.java#L56)
> 
> Did you miss to update this file?

No, it's still a valid test to check if GTK 2 is not loaded, since we don't want it to be.

**Comment by kumarabhi006** [comment]
> No, it's still a valid test to check if GTK 2 is not loaded, since we don't want it to be. It doesn't hurt anyway.

Yeah, that's true.
Although this condition check for GTK2 can be removed as it breaks out of loop and throw RTE if it is looking for GTK2.

But I guess this may not be in the scope of this fix.

File: src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java [hunk 5]
```diff
@@ -213,17 +213,6 @@ public void paintRadioButtonMenuItemBackground(SynthContext context,
         if (gtkState == SynthConstants.MOUSE_OVER) {
             if (GTKLookAndFeel.is3()) {
                 paintComponentBackground(context, g, x, y, w, h);
```
Comment (prrace) [inline src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java:215]: So do we really need the is3() check ?? Seems like not to me. Same for the case below at line 573.

Comment (azvegint) [reply src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java]: It is not the only place, I doubted it, but now updated to remove all usages.
However, prefer to keep the `is3` method itself.

**Review by kumarabhi006 — APPROVED** [review]
Changes looks good to me.
Hope you have run all the tests with latest changes.

**Review by prrace — APPROVED** [review]

**Review by magicus — APPROVED** [review]
Build changes are trivially fine.

**Pull request closed without merging**